src/GraphQLController.php

```diff
@@ -96,15 +97,20 @@ static function (RequestError $err): Error {
         }
 
         try {
-            $query = $this->handleAutomaticPersistQueries($schemaName, $params);
+            // In case of an APQ cache hit, parsedQuery contains the AST of the provided query
+            // and is subsequently used to speed up the execution.
+            [
+                'query' => $query,
+                'parsedQuery' => $parsedQuery,
+            ] = $this->handleAutomaticPersistQueries($schemaName, $params);
         } catch (AutomaticPersistedQueriesError $e) {
```

Comment:
Perhaps it is better to add error handling related to the method `Parser::parse($query)` (like GraphQL\Error\SyntaxError).
What do you think about it?

Reply:
Good point, will do!

Reply:
This is handled now + added a test, again thanks!